src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -452,18 +452,49 @@ void MinMdnsResolver::AdvancePendingResolverStates()
         else if (resolver->IsActiveOperationalParse())
         {
             MATTER_TRACE_SCOPE("Active operational delegate call", "MinMdnsResolver");
-            ResolvedNodeData nodeData;
+            ResolvedNodeData nodeResolvedData;
+            CHIP_ERROR err = CHIP_ERROR_UNINITIALIZED;
+
+            if (mActiveResolves.HasBrowseFor(chip::Dnssd::DiscoveryType::kCommissionableNode))
+            {
+                DiscoveredNodeData nodeData;
+                err = resolver->Take(nodeData);
+                if (!nodeData.Is<OperationalNodeData>())
+                {
+                    err = CHIP_ERROR_INVALID_ARGUMENT;
```

Comment:
Isn't this losing the `err` that `Take` might have returned?

Reply:
Yes, but if there is no OperationalNodeData available then even if `Take` returned success we cannot use it. If the data is of type OperationalNodeData only then the `err` from `Take` has significance. But I guess I should do that type check only is `Take` is successful. I will update the logic a bit.

Reply:
this logic was no longer needed with removal of inheritance. removed it.